### PR TITLE
RDM-8529 - external functional components integration

### PR DESCRIPTION
**Pull request by smathangi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-8529

### Change description ###
Integration for 
- IDAM authentication
- S2S authentication
- ccd-data-store-api
- PRD

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/managecase/security/JwtGrantedAuthoritiesConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @DisplayName("No Claims should return empty authorities")
+    void shouldReturnEmptyAuthoritiesWhenClaimNotAvailable() {
+        Jwt jwt = mock(Jwt.class);
```
Comment (rafalkalita) [inline src/test/java/uk/gov/hmcts/reform/managecase/security/JwtGrantedAuthoritiesConverterTest.java:39]: We create mock in all test methods. Maybe we could move it to the class level?

File: charts/aac-manage-case-assignment/values.yaml [hunk 2]
```diff
@@ -3,8 +3,15 @@ java:
...
   keyVaults:
     "aac":
       secrets:
         - AppInsightsInstrumentationKey
+        - aac-manage-case-s2s-secret
+  environment:
+    IDAM_USER_URL: https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net
+    OIDC_ISSUER: https://forgerock-am.service.core-compute-idam-{{ .Values.global.environment }}.internal:8443/openam/oauth2/hmcts
+    S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
+    MANAGE_CASE_S2S_AUTHORISED_SERVICES: some_caller
```
[the remark this answers is not in the document]
Comment (smathangi) [reply charts/aac-manage-case-assignment/values.yaml:15]: yes, for now :)

File: build.gradle [hunk 3]
```diff
@@ -173,7 +177,7 @@ dependencies {
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-aop'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-json'
-  implementation group: 'uk.gov.hmcts.reform', name: 'health-spring-boot-starter', version: '0.0.5'
```
[the remark this answers is not in the document]
Comment (smathangi) [reply build.gradle:176]: Spring boot 2.3 has now in-built liveness and readiness endpoints. No need of our custom library.

File: src/test/java/uk/gov/hmcts/reform/managecase/security/JwtGrantedAuthoritiesConverterTest.java (hunk 1 above)
Comment (smathangi) [reply src/test/java/uk/gov/hmcts/reform/managecase/security/JwtGrantedAuthoritiesConverterTest.java]: done

**Pull request merged**